src/Symfony/Component/DependencyInjection/Secrets/JweHandler.php

```diff
@@ -0,0 +1,291 @@
+            ));
+        }
+    }
+
+    /**
+     * Decrypts all populated secrets and writes the plaintext to the provided $fileLocation.
+     */
+    public function writePlaintext(string $fileLocation): void
+    {
+        $this->validateConfig($decryptRequired = true);
+        $plaintextSecrets = array();
+        foreach (array_keys($this->secrets) as $key) {
+            $plaintextSecrets[$key] = $this->decrypt($key);
+        }
+
+        $this->writeFile($fileLocation, json_encode($plaintextSecrets, self::JSON_ENCODE_OPTIONS));
+    }
+
+    /**
+     * helper method to format a sodium crypto box keypair from the configured public and private key values.
+     *
+     * @return string
+     */
+    private function collectKeyPair()
+    {
+        return sodium_crypto_box_keypair_from_secretkey_and_publickey(
+            $this->getPrivateKey(),
+            $this->getPublicKey()
+        );
+    }
+
+    private function getPrivateKey(): string
+    {
+        return $this->readKey($this->privateKeyLocation);
+    }
+
+    private function getPublicKey(): string
+    {
+        return $this->readKey($this->publicKeyLocation);
+    }
+
+    private function isValidKeyLocation(string $filePath): bool
+    {
+        return $this->isValidSecretsFile($filePath) &&
+            (SODIUM_CRYPTO_BOX_SECRETKEYBYTES === filesize($filePath));
+    }
+
+    private function isValidSecretsFile(string $filePath): bool
+    {
+        return stream_is_local($filePath) && is_readable($filePath);
+    }
+
+    /**
+     * @return array|false
+     */
+    private function populateSecrets()
+    {
+        return $this->secrets = $this->readSecrets($this->secretsLocation);
+    }
+
+    private function readKey(string $keyLocation)
+    {
+        return file_get_contents($keyLocation);
+    }
+
+    /**
+     * returns the decoded json secrets from $secretsLocation or false if the file does not contain valid json.
+     *
+     * @return array|false
+     */
+    private function readSecrets(string $secretsLocation)
+    {
+        $rawSecrets = trim(file_get_contents($secretsLocation));
+
+        return json_decode($rawSecrets, true);
+    }
+
+    private function writeEncrypted(): void
+    {
+        $this->writeFile($this->secretsLocation, json_encode($this->secrets, self::JSON_ENCODE_OPTIONS));
+    }
+
+    private function writeFile(string $fileLocation, string $content, $makeConfigDir = false): void
+    {
+        $fileDir = pathinfo($fileLocation, PATHINFO_DIRNAME);
+
+        if (!is_dir($fileDir)) {
+            //TODO make this read whatever the dev's absolute config path is and check against that
+            if ($makeConfigDir && ('config' === pathinfo($fileDir, PATHINFO_DIRNAME))) {
+                mkdir($fileDir);
+            } else {
+                throw new InvalidArgumentException(sprintf(
+                    'directory %s for encrypted secrets file location %s does not exist',
```

Comment:
I think this error isn't quite accurate - because this method could be used for a lot of different things?

Reply:
👍